code/__HELPERS/unsorted.dm

```diff
@@ -1517,3 +1517,20 @@ GLOBAL_DATUM_INIT(dview_mob, /mob/dview, new)
 	for(var/V in GLOB.player_list)
 		if(is_servant_of_ratvar(V) || isobserver(V))
 			. += V
+
+//src may be null, but it does need to be a typed var
+#define NAMEOF(datum, X) (list(##datum.##X, #X)[2])
```

Comment:
Not seeing the use case here.

Reply:
https://github.com/tgstation/tgstation/blob/master/code/controllers/globals.dm#L16
https://github.com/tgstation/tgstation/blob/master/code/controllers/subsystem/blackbox.dm#L49

Wherever we use the names of variables in code we should use NAMEOF for sanity

Reply:
I see, the list threw me off but it's just so that the define still has only a single return value which you index out, not bad.